### PR TITLE
Miri can catch panics now

### PR DESCRIPTION
**Pull request by RalfJung**
Also update some other Miri-related comments.

### CODE REVIEW

File: src/map.rs [hunk 1]
```diff
@@ -3377,7 +3377,6 @@ mod test_map {
 
         let mut m = HashMap::new();
 
-        // FIXME: https://github.com/rust-lang/miri/issues/653
         let mut rng = {
             let seed = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
             SmallRng::from_seed(seed)
```
Comment (RalfJung) [inline src/map.rs]: This issue has been closed, meaning `getrandom` now works on Miri. This is still not host randomness by default; see the [Miri README](https://github.com/rust-lang/miri/#miri--z-flags-and-environment-variables). But wouldn't it be better to keep this test deterministic? That's why I didn't change the code.

**Comment by RalfJung** [comment]
CI failures look unrelated:
```
Unable to find image 'rustembedded/cross:i686-unknown-linux-gnu-0.1.16' locally
i686-unknown-linux-gnu-0.1.16: Pulling from rustembedded/cross

Status: Downloaded newer image for rustembedded/cross:i686-unknown-linux-gnu-0.1.16
error: no such subcommand: `cargo`
```

**Comment by Amanieu** [comment]
blocked on https://github.com/rust-embedded/cross/issues/357

**Pull request merged**